### ISSUE TITLE
Consolidate `ResourceManager` methods to be connection-type agnostic

### ISSUE THREAD

**Issue by shouples** [confluentinc/vscode#430]
We have a lot of CRUD-like methods that are broken up between `CCloud*` and `Local*` that should be merged into a generic resource-specific method (e.g. `setSchemaRegistries()` instead of `setCCloudSchemaRegistries()`).

This will require generalizing the existing [`topicKeyForCluster()`](https://github.com/confluentinc/vscode/blob/main/src/storage/resourceManager.ts#L386C3-L395) method to break up the key type for general purpose usage.

Once this is done, it'll be easier to extend the types of resources (CP/remote/etc) that we are able to store.
